lib/puppet/transaction/report.rb

```diff
@@ -147,8 +147,10 @@ def as_logging_destination(&block)
   end
 
   # @api private
-  def <<(msg)
-    @logs << msg
+  def <<(log_entry)
+    # runtime errors after completion of report are added after report has finalized
+    @status = 'failed' if log_entry.level == :err
+    @logs << log_entry
```

Comment:
One possible issue is https://tickets.puppetlabs.com/browse/PUP-1763. Currently, puppet can fail to pluginsync (because there isn't al least one module with a `lib` directory), and I think puppet will emit an error message as a result. But I think this change will make the report fail even though the agent run did succeed...